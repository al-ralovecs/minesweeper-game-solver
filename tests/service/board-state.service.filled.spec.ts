import { disposition } from '../fixtures/state/3x3.filled.json';
import BoardDto from '../../src/dto/board.dto';
import BoardStateService from '../../src/service/board-state.service';

describe('BoardStateService', () => {
    test('check how computes filled board', () => {
        const board = new BoardDto(disposition);
        const boardStateService = new BoardStateService(board.height, board.width, 3);

        boardStateService.setBoard = board;
        boardStateService.process();

        expect(boardStateService.getBoardState).toMatchSnapshot();
    })
});
