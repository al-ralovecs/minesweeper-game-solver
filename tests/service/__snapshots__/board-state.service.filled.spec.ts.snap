// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardStateService check how computes filled board 1`] = `
BoardStateDto {
  "action": Array [
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
  ],
  "actionList": Array [],
  "adjFlagsConfirmed": Array [
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
  ],
  "adjFlagsOnBoard": Array [
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
    ],
  ],
  "adjUnrevealed": Array [
    Array [
      1,
      4,
      2,
    ],
    Array [
      3,
      5,
      4,
    ],
    Array [
      2,
      3,
      3,
    ],
  ],
  "adjacentLocations1": Array [
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
  ],
  "board": Array [
    Array [
      -1,
      1,
      -1,
    ],
    Array [
      2,
      -1,
      -1,
    ],
    Array [
      2,
      -1,
      -1,
    ],
  ],
  "chordLocations": Array [],
  "expectedTotalMines": 3,
  "flagConfirmed": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "flagOnBoard": Array [
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
    Array [
      false,
      false,
      false,
    ],
  ],
  "height": 3,
  "livingWitnesses": LocationSetDto {
    "data": Array [
      LocationDto {
        "x": 1,
        "y": 0,
      },
      LocationDto {
        "x": 0,
        "y": 1,
      },
      LocationDto {
        "x": 0,
        "y": 2,
      },
    ],
  },
  "numOfHidden": 6,
  "revealed": Array [
    Array [
      false,
      true,
      false,
    ],
    Array [
      true,
      false,
      false,
    ],
    Array [
      true,
      false,
      false,
    ],
  ],
  "totalFlags": 0,
  "totalFlagsConfirmed": 0,
  "width": 3,
}
`;
